worlds/smz3/TotalSMZ3/Regions/Zelda/PalaceOfDarkness.py

```diff
@@ -29,7 +29,7 @@ def __init__(self, world, config: Config):
             Location(self, 256+127, 0x1EA43, LocationType.Regular, "Palace of Darkness - Compass Chest",
                 lambda items: items.KeyPD >= (4 if (items.Hammer and items.Bow and items.Lamp) or config.Keysanity else 3)),
             Location(self, 256+128, 0x1EA46, LocationType.Regular, "Palace of Darkness - Harmless Hellway",
-                lambda items: items.KeyPD >= (4 if (items.Hammer and items.Bow and items.Lamp) or config.Keysanity else 3 if 
+                lambda items: items.KeyPD >= ((4 if (items.Hammer and items.Bow and items.Lamp) or config.Keysanity else 3) if 
                         self.GetLocation("Palace of Darkness - Harmless Hellway").ItemIs(ItemType.KeyPD, self.world) else
                         6 if (items.Hammer and items.Bow and items.Lamp) or config.Keysanity else 5))
```

Comment:
?: evaluates from left to right, so it's easier to read than python's else/if, but in all honesty the original code is prone to errors as well and should have used more parentheses to signal it's two levels of Ifs. `A ? (B ? c : d) : (B ? e : f)` may have survived to port to python.

I would suggest adding another set of parentheses around `6 if (items.Hammer and items.Bow and items.Lamp) or config.Keysanity else 5` to better show it's two levels. Otherwise the fix seems good. Trying to stay within 120 chars/line makes it look a bit awkward again

```suggestion
                lambda items: items.KeyPD >= (
                        (4 if (items.Hammer and items.Bow and items.Lamp) or config.Keysanity else 3) if
                        self.GetLocation("Palace of Darkness - Harmless Hellway").ItemIs(ItemType.KeyPD, self.world)
                        else (6 if (items.Hammer and items.Bow and items.Lamp) or config.Keysanity else 5)))
```